services/slam/builtin/builtin.go

```diff
@@ -369,7 +371,32 @@ func (slamSvc *builtIn) Position(ctx context.Context, name string) (*referencefr
 		return nil, errors.Wrap(err, "error getting SLAM position")
 	}
 
-	return referenceframe.ProtobufToPoseInFrame(resp.Pose), nil
+	pInFrame := referenceframe.ProtobufToPoseInFrame(resp.Pose)
+
+	// TODO DATA-531: Remove extraction and conversion of quaternion from the extra field in the response once the Rust SDK
+	// is available and the desired math can be implemented on the orbSLAM side
+	extra := resp.Extra.AsMap()
+
+	if val, ok := extra["quat"]; ok {
+		q := val.(map[string]interface{})
+
+		valTheta, ok1 := q["otheta"]
+		valX, ok2 := q["ox"]
+		valY, ok3 := q["oy"]
+		valZ, ok4 := q["oz"]
```

Comment:
Is this definitely a quaternion? It looks like it is formatted as an R4 Axis Angle

Reply:
old code with non-updated var names